tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,301 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Castle.Core.Internal;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Http;
+using Microsoft.Identity.Json;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.Identity.Test.Integration.NetFx.Infrastructure;
+using Microsoft.Identity.Test.LabInfrastructure;
+using Microsoft.Identity.Test.Unit;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NSubstitute.Exceptions;
+
+namespace Microsoft.Identity.Test.Integration.HeadlessTests
+{
+    [TestClass]
+    public class ManagedIdentityTests
+    {
+        private static readonly string[] s_msi_scopes = { "https://management.azure.com" };
+        private static readonly string[] s_wrong_msi_scopes = { "https://managements.azure.com" };
+        private static readonly string s_clientId = "client_id";
+
+        //http proxy base URL 
+        private static readonly string s_baseURL = "https://service.msidlab.com/";
+
+        //Shared User Assigned Client ID
+        private const string UserAssignedClientID = "3b57c42c-3201-4295-ae27-d6baec5b7027";
+
+        //Non Existent User Assigned Client ID 
+        private const string NonExistentUserAssignedClientID = "3b57c42c-9999-9999-zz99-d6baec5b7027";
+
+        //Error Messages
+        private const string UserAssignedIdDoesNotExist = "[Managed Identity] Error Message: No User Assigned or Delegated Managed Identity found for specified ClientId/ResourceId/PrincipalId.";
+
+        //Resource ID of the User Assigned Identity 
+        private const string UamiResourceId = "/subscriptions/c1686c51-b717-4fe0-9af3-24a20a41fb0c/" +
+            "resourcegroups/MSAL_MSI/providers/Microsoft.ManagedIdentity/userAssignedIdentities/" +
+            "MSAL_MSI_USERID";
+
+        //non existent Resource ID of the User Assigned Identity 
+        private const string Non_Existent_UamiResourceId = "/subscriptions/userAssignedIdentities/NO_ID";
+
+        [TestMethod]
+        public async Task ManagedIdentity_WithoutEnvironmentVariables_ThrowsAsync()
+        {
+            //Arrange
+            string result = string.Empty;
+
+            IConfidentialClientApplication cca = CreateCCAWithProxy(s_baseURL);
+
+            MsalServiceException ex = await AssertException.TaskThrowsAsync<MsalServiceException>(async () =>
```

Comment:
I don't think this will be MsalServiceException. When there is no environment variable, IMDS source is created and tries to acquire token from IMDS endpoint which results in HttpRequestException ` System.Net.Http.HttpRequestException: A socket operation was attempted to an unreachable network.`

Reply:
will need to test this a bit more but I do get a MSALServiceException

Reply:
modified to HttpRequestException, the problem was going through the proxy. thanks @neha-bhargava